app.py

```diff
@@ -15,9 +15,108 @@
 # You should have received a copy of the GNU General Public License
 # along with this program. If not, see <http://www.gnu.org/licenses/>.
 
-"""This is the main script of the template project."""
+"""This is the main script of Thoth SLO reporter.
+
+    Thanks to DataHub Team in the Red Hat AICoE!!
+"""
+
+import os
+import logging
+import smtplib
+import datetime
+import pandas as pd
+
+from prometheus_api_client import Metric, MetricsList, PrometheusConnect
+from prometheus_api_client.utils import parse_datetime, parse_timedelta
+from prometheus_client import CollectorRegistry, Gauge, push_to_gateway
+
+from email.mime.multipart import MIMEMultipart
+from email.mime.text import MIMEText
+
+
+_LOGGER = logging.getLogger("thoth.slo_reporter")
+
+
+def push_thoth_sli_weekly_metric(
+    weekly_value_metric: Metric, prometheus_registry: CollectorRegistry, pushgateway_endpoint: str
+):
+    """Push Thoth SLI weekly metric to PushGateway."""
+    weekly_metric = Gauge("thoth_sli_weekly", "Weekly Thoth Service Level Indicators", registry=prometheus_registry)
+
+    weekly_value_metric = float(weekly_value_metric[0]["value"][1])
+    weekly_metric.set(weekly_value_metric)
+
+    push_to_gateway(pushgateway_endpoint, job="Weekly Thoth SLI", registry=prometheus_registry)
+
+    return weekly_value_metric
+
+
+def generate_email(learned_packages: int):
+    """General email to be sent."""
+    end_time = datetime.datetime.now()
+    start_time = end_time - datetime.timedelta(days=7)
+    start_time_epoch = int(start_time.timestamp() * 1000)
+    end_time_epoch = int(end_time.timestamp() * 1000)
+
+    # TODO: Show one single dashboard for all metrics to be shown.
+    grafana_dashboard_url = (
+        f"https://grafana.datahub.redhat.com/dashboard/db/thoth-knowledge-graph-content-metrics-stage?"
+        + f"refresh=1m&panelId=23&fullscreen&orgId=1&from={start_time_epoch}&to={end_time_epoch}"
+    )
+
+    return MIMEText(
+        f"<strong>Thoth Solved {learned_packages} Python Packages in the last week.</strong> \
+            <br><br> \
+            Python Packages Information collected in the last week by Thoth. \
+            <br> \
+            The following dashboard panel contains Python Packages info collected: <a href='{grafana_dashboard_url}'> Python Packages Info.</a> \
+            <br>",
+        "html",
+    )
+
+
+def send_sli_email(server: str, sender_address: str, recipients: str, learned_packages: int):
+    """Send email about Thoth Service Level Objectives."""
+    _MAIL_SERVER = smtplib.SMTP(server)
+
+    msg = MIMEMultipart()
+    msg["Subject"] = "Thoth Weekly Service Level Indicators"
```

Comment:
I would put week number and year or so to the subject so it's easily differentiable in search results. WDYT?

Reply:
Yes sounds good :)